libbeat/cmd/instance/metrics.go

```diff
@@ -36,3 +59,35 @@ func reportInfo(_ monitoring.Mode, V monitoring.Visitor) {
 	uptime := int64(delta.Seconds())*1000 + int64(delta.Nanoseconds()/1000000)
 	monitoring.ReportInt(V, "uptime.ms", uptime)
 }
+
+func reportCPU(_ monitoring.Mode, V monitoring.Visitor) {
+	V.OnRegistryStart()
+	defer V.OnRegistryFinished()
+
+	cpuUsage, normalizedCPU := getCPUUsage()
+	monitoring.ReportFloat(V, "usage", cpuUsage)
+	monitoring.ReportFloat(V, "usage.normalized", normalizedCPU)
+}
+
+func getCPUUsage() (float64, float64) {
+	pid := os.Getpid()
+
+	sample := cpuSample{
+		time:      time.Now(),
+		procTimes: sigar.ProcTime{},
+	}
+
+	if err := sample.procTimes.Get(pid); err != nil {
+		return 0, 0
+	}
+
+	dTime := sample.time.Sub(lastSample.time)
+	dMilli := dTime / time.Millisecond
+	dCPU := int64(sample.procTimes.Total - lastSample.procTimes.Total)
```

Comment:
If we would only report `sample.procTimes.Total` with a timestamp + # of cores, we could leave the calculation of 1, 5 and 15 minutes average to Kibana? Like this it would be up to KB to decide which averages it wants to calculate. Perhaps I miss something?

Reply:
I like your idea. @tsullivan WDYT?

Reply:
Not sure I'm following... is this for system load average?

cc @pickypg

Reply:
Right now it returns the number of CPU usage between the last two sampling times. Let's say that the sampling time is 1 minute. It would return the following metrics in case of constant usage:
```
beat.cpu.usage=0.1   // at 12:00
beat.cpu.usage=0.1   // at 12:01
beat.cpu.usage=0.1   // at 12:02
```

@ruflin's idea is to return a timestamp and the total CPU usage since the Beat has been started. It would look like this:
```
timestamp=12:00 beat.cpu.usage=0.1
timestamp=12:01 beat.cpu.usage=0.2
timestamp=12:02 beat.cpu.usage=0.3 
```
@ruflin please, correct me if I misunderstood your idea.

Reply:
Exactly. So on the UI side there would be fully flexibility if it should be 1,5,15 or 1,10,30 minute average and we report only one value each time.

Reply:
A potential problem came to my mind. It might be confusing to users to see that the internal metrics are reported differently from metricbeat's format. For the sake of consistency, I would stick with the current solution. (Even if it's not as flexible as your proposal is.)

Reply:
So my only concern with this approach is that it's different from what we're currently doing.

In the short term, can we have both? The other parts of the stack also report it in the not-rolled-up format, so we it may be confusing to present a different, albeit better approach to showing the utilization. However, longer term, I would love to see us move toward the more flexible approach, which is only possible if we already have the data (and it seems cheap to have both).

Reply:
Having both SGTM